packages/hadron-document/src/element.ts

```diff
@@ -840,6 +850,47 @@ export class Element extends EventEmitter {
     }
   }
 
+  getVisibleElements() {
+    if (!this.elements || !this.expanded) {
+      return [];
+    }
+    return [...this.elements].slice(0, this.maxVisibleElementsCount);
+  }
+
+  setMaxVisibleElementsCount(newCount: number) {
+    if (!this._isExpandable(this.originalExpandableValue)) {
```

Comment:
We updated the `_isExpandable` usage in https://github.com/mongodb-js/compass/pull/6013
There is a thread on how it can be better used there. Now it's less of a static method.
```suggestion
    if (!this._isExpandable()) {
```
I'm thinking the recent changes are preventing a bug here where someone could be editing a sub document with a lot of fields in it and change the type.

Thanks for the code comment on the difference between the `25` when editing and `1000` when not editing.

Reply:
Thanks for catching this Rhys. I took a pull from main but forgot to update this. Just did that.